### PR TITLE
Add query to set default filter for worktray

### PR DESCRIPTION
**Pull request by georgeschena**
### WHAT: 
Added a query to filter the default view of the work tray
### WHY: 
So not to show no_action in the immediate action tab

### CODE REVIEW

File: spec/lib/hackney/income/stored_tenancies_gateway_spec.rb [hunk 1]
```diff
@@ -393,23 +393,29 @@
       context 'with is_paused set false' do
         let(:is_paused) { false }
 
-        it 'shows the number pages of of paused cases' do
+        it 'shows the number pages of paused cases' do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/stored_tenancies_gateway_spec.rb]: Should this be 
```suggestion
        it 'shows the number of pages of paused cases' do
```
?

File: spec/lib/hackney/income/stored_tenancies_gateway_spec.rb [hunk 2]
```diff
@@ -393,23 +393,29 @@
...
 
       context 'with is_paused set true' do
         let(:is_paused) { true }
 
-        it 'shows the number pages of of paused cases' do
+        it 'shows the number pages of paused cases' do
           expect(subject).to eq(expected_num_pages(num_paused_cases, num_pages))
         end
+
+        it 'shows the number pages of paused cases with one no_action classification' do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/stored_tenancies_gateway_spec.rb]: This setup is inside the `it`, compared to the rest of this spec where it's set up as part of context blocks. I worry someone might miss it in the future because of that. Is there a way to pull it up and make it more visible?

File: spec/lib/hackney/income/stored_tenancies_gateway_spec.rb [hunk 3]
```diff
@@ -449,8 +455,10 @@
 
       let(:classification) { nil }
 
-      it 'returns all tenancies' do
-        expect(subject.count).to eq(cases_with_no_action + cases_with_warning_letter_action)
+      context 'when the classification param is nil' do
+        it 'does not return :no_action tenancies' do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/stored_tenancies_gateway_spec.rb]: The `it` description doesn't seem to match the assertion here. I'm having trouble following what the assertion is, even (`subject` is pretty distant to this spec). 

Can the `it` block refer to something positively rather than negatively? e.g. "returns only X tenancies"? I think that would be clearer.

File: lib/hackney/income/stored_tenancies_gateway.rb [hunk 4]
```diff
@@ -69,7 +69,11 @@ def number_of_pages_for_user(user_id:, number_per_page:, is_paused: nil, classif
       def tenancy_filtered_by_paused_state_for(user_id, is_paused, classification)
         query = GatewayModel.where('assigned_user_id = ? AND balance > ?', user_id, 0)
 
-        query = query.where(classification: classification) if classification
+        if classification
+          query = query.where(classification: classification)
+        elsif !is_paused
+          query = query.where.not(classification: :no_action).or(query.where(classification: nil))
+        end
```
Comment (kjdchapman) [inline lib/hackney/income/stored_tenancies_gateway.rb]: This was a little hard to parse. I'm not sure how to write it better with a brief look - maybe extracting methods to give semantic meaning to each of these where clauses?:

```
add_classification_filter(query) if classification.present?
add_all_actions_filter(query) unless is_paused
```

Not sure if ^ is best way, but think it needs a little love.

**Comment by richrace** [comment]
We're still working on this. This was meant to be a draft PR. Whoops. 😅

File: lib/hackney/income/stored_tenancies_gateway.rb [hunk 5]
```diff
@@ -77,7 +77,12 @@ def tenancies_filtered_for(user_id, filters)
           balance > ?', user_id, 0)
 
         query = query.where('patch_code = ?', filters[:patch]) if filters[:patch]
-        query = query.where(classification: filters[:classification]) if filters[:classification]
+
+        if filters[:classification]
+          query = query.where(classification: filters[:classification])
+        elsif !filters[:is_paused]
+          query = query.where.not(classification: :no_action).or(query.where(classification: nil))
```
Comment (elena-vi) [inline lib/hackney/income/stored_tenancies_gateway.rb]: I find this `or` statement incredibly confusing

Comment (georgeschena) [reply lib/hackney/income/stored_tenancies_gateway.rb]: Agreed

File: lib/hackney/income/stored_tenancies_gateway.rb (hunk 4 above)
Comment (richrace) [reply lib/hackney/income/stored_tenancies_gateway.rb]: Updated the existing code. 

But we want to refactor this area to be a Query Object at some point.

**Comment by richrace** [comment]
:eyes: Ready! :tada:

File: spec/lib/hackney/income/stored_tenancies_gateway_spec.rb [hunk 6]
```diff
@@ -397,23 +397,33 @@
...
+        it 'shows the number of pages of paused cases' do
           expect(subject).to eq(expected_num_pages(num_active_cases, num_pages))
         end
       end
 
       context 'with is_paused set true' do
         let(:is_paused) { true }
 
-        it 'shows the number pages of of paused cases' do
+        it 'shows the number pages of paused cases' do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/stored_tenancies_gateway_spec.rb]:
```suggestion
        it 'shows the number of pages of paused cases' do
```

File: spec/lib/hackney/income/stored_tenancies_gateway_spec.rb [hunk 7]
```diff
@@ -397,23 +397,33 @@
...
           expect(subject).to eq(expected_num_pages(num_paused_cases, num_pages))
         end
+
+        context 'with one no_action classification case' do
+          before do
+            create(:case_priority, assigned_user_id: user.id, balance: 40, is_paused_until: Faker::Date.forward(1), classification: :no_action)
+          end
+
+          it 'shows the number pages of paused cases with one no_action classification' do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/stored_tenancies_gateway_spec.rb]:
```suggestion
          it 'shows the number of pages of paused cases with one no_action classification' do
```

File: spec/lib/hackney/income/stored_tenancies_gateway_spec.rb [hunk 8]
```diff
@@ -397,23 +397,33 @@
...
+            expect(subject).to eq(expected_num_pages(num_paused_cases + 1, num_pages))
+          end
+        end
       end
 
       context 'with is_paused not set' do
         let(:is_paused) { nil }
 
-        it 'shows the number pages of of paused cases' do
+        it 'shows the number pages of paused cases' do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/stored_tenancies_gateway_spec.rb]:
```suggestion
        it 'shows the number of pages of paused cases' do
```
